RIOT-OS/RIOT
# Boards/Generic support for the ATSAMR21G18-MR210UA module

### Contribution description
This commit adds initial support for the [ATSAMR21 Breakout Board](http://hackerspace.teubi.co/wiki/index.php?title=HSSV_ATSAMR21_Breakout_English) from [Hackerspace San Salvador](http://hackerspace.teubi.co/).

The initial support includes:
- Board definition for the [ATSAMR21G18-MR210UA module](http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-42475-ATSAMR21G18-MR210UA_Datasheet.pdf)
- Patch for using bossac to load RIOT to the board via USB.
- Support for antenna diversity

Features not included yet:
- Support for using the embedded MAC address
- Drivers for the included 4Mb flash memory
- Drivers for the crypto-auth chip
- USB-CDC support to remove the requirement of an external USB-Serial converter

Design files for this board are available over: [https://github.com/hackerspacesv/RIOT-hardware](https://github.com/hackerspacesv/RIOT-hardware)

**Note:** This is an OSHW board, we would like to have support for RIOT before releasing the first production batch. However we have the initial testing batch and we can provide a couple of them to the RIOT community for testing purposes. We also think that other ports using the ATSAMR21 could benefit from this board definition as an example.

### Bootloader info
This board supports the Arduino MRK1000 bootloader without any troubles. However we recommend to use our port. The port uses our reserved [VID/PID USB identifiers](http://pid.codes/1209/053A/) from [pid.codes](http://pid.codes/).

Procedure:
Clone our ArduinoCore-samd port:
`git clone https://github.com/hackerspacesv/ArduinoCore-samd`

You can find the pre-compiled bootloader under the directory:
`cd ArduinoCore-samd/bootloaders/HSSV_SAMR21_Mote`

'Burn' the bootloader using your preffered JTAG programmer. Bin/hex files are available under the following names:
`samr21_sam_ba_HSSV_SAMR21_Mote.bin`
`samr21_sam_ba_HSSV_SAMR21_Mote.hex`

### Testing procedure
Clone the RIOT repository from Hackerspace San Salvador with the following command: 
`git clone https://github.com/hackerspacesv/RIOT`

Change to the network examples directory:
`cd RIOT/examples/gnrc_networking`

Compile the example specifying the breakout board:
`make BOARD=hssv-atsamr21-breakout`

Execute the following command to flash the board:
`make flash BOARD=hssv-atsamr21-breakout`

Connect an external USB-Serial bridge and open a terminal client with the following command (We assume minicom and the serial device /dev/ttyUSB0):
`minicom -b 115200 -D /dev/ttyUSB0`

If you press the reset button you'll see a message as the following:
> main(): This is RIOT! (...)
> RIOT Network stack example application
> All up, running the shell now
> >

### Issues/PRs references
N/A
